### PR TITLE
Algo identification in less then 10 lines of code

### PR DESCRIPTION
**Pull request by papampi**
LukePicchi idea for algorithm identification

### CODE REVIEW

**Comment by papampi** [comment]
@LuKePicci couldn't wait for your magic, had to see if I can do it myself...

**Comment by LuKePicci** [comment]
AHAHAH seems legit

File: 0algo_id [hunk 1]
```diff
@@ -1,70 +1,15 @@
...
-elif [[ $COIN == ONION ]]
-then
-  ALGO=X13
-elif [[ $COIN == ZPOOL_SKEIN || $COIN == MPH_SKEIN ]]
-then
-  ALGO=SKEIN
-else
-  ALGO=UNDEFINED
-fi
+ALGOS="ETHASH EQUIHASH"
```
Comment (LuKePicci) [inline 0algo_id]: Ok, remember to add other algos here

File: 0algo_id [hunk 2]
```diff
@@ -1,70 +1,15 @@
...
+
+for ALGO in $ALGOS
```
Comment (LuKePicci) [inline 0algo_id]: rename the loop variable in something different, simply 'a' should be fine

File: 0algo_id [hunk 3]
```diff
@@ -1,70 +1,15 @@
...
+do
+  eval 'ALGO_COINS=${ALGO}_COINS'
+  if [[ "${!ALGO_COINS}"  =~ "$COIN" ]]
```
Comment (LuKePicci) [inline 0algo_id]: You should not need the eval if you use ${!${a}_COINS}

**Comment by papampi** [comment]
@LuKePicci Do we need UNDEFINED algo?

**Comment by LuKePicci** [comment]
> @LuKePicci Do we need UNDEFINED algo?

Yes, its needed for switchers that do not edit 1bash with the new mining coin

**Comment by papampi** [comment]
But 0miner has no option for undefined and miner will not start unless it identifies the algo

**Comment by papampi** [comment]
Ooops, yup , salfter switchers

**Comment by papampi** [comment]
How do you suggest it. I tried but couldn't add undefined.

**Comment by LuKePicci** [comment]
You have to set ALGO="UNDEFINED" before the loop, you probably had problems because you were using ALGO as both env variable and loop local.

File: 0algo_id (hunk 3 above)
Comment (papampi) [reply 0algo_id]: If I remove eval and use ${!${ALGO}_COINS} it gives error.
test.sh: line 23: ${!${ALGO}_COINS}: bad substitution

Comment (LuKePicci) [reply 0algo_id]: Will check later, now I have to go away

**Comment by papampi** [comment]
All should be ok now.

**Review by LuKePicci — APPROVED** [review]

**Pull request merged**